### PR TITLE
Updated Code samples to AWS SDK 2.x

### PR DESCRIPTION
**Pull request by arunx2**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by arunx2** [comment]
Can we merge this PR?

**Comment by dblock** [comment]
Came here to update the same docs! I think we should stop copy-pasting `AwsRequestSigningApacheInterceptor` code. The one in the attached zip has a few bugs (e.g. https://github.com/acm19/aws-request-signing-apache-interceptor/issues/38). 

There's now a maintained community fork that is AWS SDK 2.0 and has been released to maven: https://github.com/acm19/aws-request-signing-apache-interceptor.

How about we point to it?

I've also raised to the AWS SDK team the possibility of bringing the fork back under an AWS org.

**Comment by lizsnyder** [comment]
Hi @arunx2, I'm so sorry that I missed this. For some reason I wasn't receiving notifications of this change, so I didn't see it until someone pointed it out to me today. I will test this out tomorrow to make sure it works, and then merge.

**Comment by lizsnyder** [comment]
@arunx2 do you have any comment on Daniel's comment above?

**Comment by arunx2** [comment]
@dblock I'll exclude the code and update the zip file. @lizsnyder You can review after the changes with the code.

**Comment by arunx2** [comment]
@dblock While I like the idea of not copying the code, I'm little skeptical to add a dependency not managed by AWS with an official documentation
<dependency>
            <groupId>io.github.acm19</groupId>
            <artifactId>aws-request-signing-apache-interceptor</artifactId>
            <version>2.1.1</version>
          </dependency>

I totally agree with your #3 and #4.

**Comment by dblock** [comment]
I think we should reference acm19.* because 1) that's a supported open-source package, just like org.apache.http.* or org.opensearch.action.index.*, 2) customers reading the documentation will not be copy-pasting eventually outdated, potentially buggy, or worse, insecure code. 

@lizsnyder Do you have any objections?

**Comment by lizsnyder** [comment]
> I think we should reference acm19.* because 1) that's an open-source package just like org.apache.http.* or org.opensearch.action.index.* which aren't managed by AWS either, 2) customers reading the documentation will not be copy-pasting eventually outdated, potentially buggy, or worse, insecure code.
> 
> Can we expand the zip in GitHub, so developer can see code changes, and replace the link to the zip to a link to GitHub? A zip is very developer unfriendly, we can't see history, etc.
> 
> @lizsnyder Do you have any objections?

Sure, all of that sounds great to me

**Comment by dblock** [comment]
> @dblock I totally agree with samples in a GitHub repository than the zip file. Do we have GitHub for this sample?

Can we just use this repo? sample_code/java/amazon-opensearch-docs-sample-client/...? Just not as a zip.
 
> @lizsnyder I've a hesitation to refer this acm19* in official documentation and in my opinion, org.apache.* and org.opensearch.* is backed by a large community and vetted process to get some changes into them. I'm not very familiar with this acm19.* package and not sure if this should be referred in AWS documentation. What's your take?

Note that maven artifacts can't be removed from maven central. 

And FWIW, I am a co-maintainer on the acm19 project. If it were to disappear I'd fork it back into opensearch-project at a minimum.

**Comment by lizsnyder** [comment]
Unfortunately I have to agree with @arunx2. It's not an official AWS library, so recommending it would be tough. Looking at this a little more, I think it's best to wait for the AWS SDK team to bring the fork back under an AWS org. Thoughts? Can we expedite that somehow?

**Comment by dblock** [comment]
I don't think we should expect to anything to happen with the AWS SDK team, nor should we wait for it to correct the product documentation that is currently obsolete or has bugs.  

@arunx2 want to update this PR with whichever state you think is right based on this discussion and we'll try to see what @lizsnyder is comfortable with merging?

File: doc_source/request-signing.md [hunk 1]
```diff
@@ -75,29 +76,28 @@ public class AmazonOpenSearchServiceSample {
...
-        HttpRequestInterceptor interceptor = new AWSRequestSigningApacheInterceptor(serviceName, signer, credentialsProvider);
+        Aws4Signer signer = Aws4Signer.create();
+        HttpRequestInterceptor interceptor = new AwsRequestSigningApacheInterceptor(serviceName, signer, credentialsProvider,region);
         return RestClient.builder(HttpHost.create(host)).setHttpClientConfigCallback(hacb -> hacb.addInterceptorLast(interceptor)).build();
     }
 }
 ```
 
-If you prefer the high\-level REST client, which offers most of the same features and simpler code, try the following sample, which also uses the [Amazon Web Services Request Signing Interceptor](https://github.com/awslabs/aws-request-signing-apache-interceptor):
+If you prefer the high\-level REST client, which offers most of the same features and simpler code, try the following sample, which also uses the [Amazon Web Services Request Signing Interceptor](https://github.com/awsdocs/aws-request-signing-apache-interceptor):
```
Comment (dblock) [inline doc_source/request-signing.md:86]: At the very least this link needs to be removed. The code in that repo is old and should not be used.

**Comment by lizsnyder** [comment]
> I don't think we should expect to anything to happen with the AWS SDK team, nor should we wait for it to correct the product documentation that is currently obsolete or has bugs.
> 
> @arunx2 want to update this PR with whichever state you think is right based on this discussion and we'll try to see what @lizsnyder is comfortable with merging?

Sounds good to me, thanks.

**Comment by arunx2** [comment]
Apologies for the delay. I inherited the bug fix to the sample code and referred source code instead of zip file.

**Review by dblock — APPROVED** [review]
This looks good enough to me!

I do think we should put SPDX-like copyright headers https://github.com/acm19/aws-request-signing-apache-interceptor/blob/master/src/main/java/io/github/acm19/aws/interceptor/http/AwsRequestSigningApacheInterceptor.java#L1 on all Java files here, but NBD.

**Comment by lizsnyder** [comment]
You guys are awesome! Let me take a day to review/test everything and will hopefully be able to merge tomorrow.

**Comment by lizsnyder** [comment]
Merged. Changes should show up in the public docs in about an hour or two.

**Pull request merged**